common4j/build.gradle

```diff
@@ -24,6 +25,106 @@ repositories {
     mavenCentral()
 }
 
+apply from: './versioning/version_tasks.gradle'
+
+project.ext.vstsUsername = System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_USERNAME") != null ? System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_USERNAME") : project.findProperty("vstsUsername")
+project.ext.vstsPassword = System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_ACCESSTOKEN") != null ? System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_ACCESSTOKEN") : project.findProperty("vstsMavenAccessToken")
+
+version = getAppVersionName()
+
+task sourcesJar(type: Jar) {
+    from sourceSets.main.java.srcDirs
+    classifier = 'sources'
+    destinationDirectory = reporting.file("$project.buildDir/libs/")
+}
+
+// Task to generate javadoc
+task generateJavadoc(type: Javadoc) {
+    failOnError false
+    title = "Microsoft Identity Common Java"
+    source = sourceSets.main.java
+    classpath += project.sourceSets.main.compileClasspath
+
+    options.memberLevel = JavadocMemberLevel.PUBLIC
+    options.addStringOption('Xdoclint:none', '-quiet')
+
+    exclude '**/BuildConfig.Java'
+    exclude '**/R.java'
+    destinationDir = reporting.file("$project.buildDir/javadoc")
```

Comment:
$project.buildDir/[OUTPUT_FOLDER]/javadoc/?

Reply:
(OUTPUT_FOLDER is now libs, but I also left a suggestion to move this under output/)

Reply:
I'm inclined to leave this one as is, but I'll move the jar (javadocJar and sourcesJar) to the output folder

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1363/commits/5c5b058dbc4339d9ca3aeb869e21a68dd2189ca7